Composer/packages/client/src/pages/botProject/ABSChannels.tsx

```diff
@@ -0,0 +1,674 @@
+              type: 'Microsoft.BotService/botServices/channels',
+              location: 'global',
+              properties: {
+                properties: {
+                  cognitiveServiceRegion: null,
+                  cognitiveServiceSubscriptionKey: null,
+                  isEnabled: true,
+                  customVoiceDeploymentId: '',
+                  customSpeechModelId: '',
+                  isDefaultBotForCogSvcAccount: false,
+                },
+                channelName: 'DirectLineSpeechChannel',
+                location: 'global',
+              },
+            };
+        }
+        await httpClient.put(url, data, { headers: { Authorization: `Bearer ${token}` } });
+
+        // success!!
+        setChannelStatus({
+          ...channelStatus,
+          [channelId]: {
+            enabled: true,
+            loading: false,
+          },
+        });
+
+        return {
+          enabled: true,
+          loading: false,
+        };
+      } catch (err) {
+        switch (err?.response.data?.error.code) {
+          case 'AuthenticationFailed':
+            // the auth failed for some reason.
+            break;
+          case 'ResourceGroupNotFound':
+            // this resource group is not found - in other words, can't find a channel registration in the expected spot.
+            break;
+          case 'SubscriptionNotFound':
+            // the subscription is not found or invalid
+            break;
+          default:
+            // handle error.
+            break;
+        }
+        throw new Error(err?.response.data?.error.message || 'Failed to create new channel');
+      }
+    }
+  };
+
+  const deleteChannelService = async (channelId: string) => {
+    if (currentResource) {
+      try {
+        const url = `https://management.azure.com/subscriptions/${
+          currentResource.subscriptionId || currentResource.alternateSubscriptionId
+        }/resourceGroups/${currentResource?.resourceGroupName}/providers/Microsoft.BotService/botServices/${
+          currentResource?.resourceName
+        }/channels/${channelId}?api-version=2020-06-02`;
+        await httpClient.delete(url, { headers: { Authorization: `Bearer ${token}` } });
+
+        // success!!
+        setChannelStatus({
+          ...channelStatus,
+          [channelId]: {
+            enabled: false,
+            loading: false,
+          },
+        });
+      } catch (err) {
+        switch (err?.response.data?.error.code) {
+          case 'AuthenticationFailed':
+            // the auth failed for some reason.
+            break;
+
+          case 'ResourceGroupNotFound':
+            // this resource group is not found - in other words, can't find a channel registration in the expected spot.
+            break;
+
+          case 'SubscriptionNotFound':
+            // the subscription is not found or invalid
+            break;
+
+          default:
+            // handle error.
+            break;
```

Comment:
With all this error handling, it might be easier to refactor this all into its own function.

Reply:
true!